### PR TITLE
change datediff to timestamp_diff

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
**What change does this PR introduce?** 
Updated `datediff` to `timestamp_diff` following naming changes from the timestamp_diff macro upgrade

**If this PR introduces a new macro, how did you test the new macro?** 

**If this PR introduces a modification to an existing macro, which packages is the macro currently present in and what steps were taken to test compatibility across packages?** 
tbd on testing it

**Did you update the README to reflect the macro addition/modifications?** 

- [ ] Yes
- [x] No (no logic changes, just renaming)

### CODE REVIEW

File: macros/timestamp_diff.sql [hunk 1]
```diff
@@ -24,35 +24,35 @@
...
 {% endmacro %}
 
-{% macro postgres__datediff(first_date, second_date, datepart) %}
+{% macro postgres__timestamp_diff(first_date, second_date, datepart) %}
 
     {% if datepart == 'year' %}
         (date_part('year', ({{second_date}})::date) - date_part('year', ({{first_date}})::date))
     {% elif datepart == 'quarter' %}
-        ({{ dbt_utils.datediff(first_date, second_date, 'year') }} * 4 + date_part('quarter', ({{second_date}})::date) - date_part('quarter', ({{first_date}})::date))
+        ({{ dbt_utils.timestamp_diff(first_date, second_date, 'year') }} * 4 + date_part('quarter', ({{second_date}})::date) - date_part('quarter', ({{first_date}})::date))
```
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply macros/timestamp_diff.sql]: Oh I see, sorry for the confusion. Just fixed it

File: macros/timestamp_diff.sql [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {% macro timestamp_diff(first_date, second_date, datepart) %}
-  {{ adapter.dispatch('datediff', packages = fivetran_utils._get_utils_namespaces())(first_date, second_date, datepart) }}
+  {{ adapter.dispatch('timestamp_diff', packages = fivetran_utils._get_utils_namespaces())(first_date, second_date, datepart) }}
 {% endmacro %}
 
 
 {% macro default__timestamp_diff(first_date, second_date, datepart) %}
 
-    datediff(
+    timestamp_diff(
```
Comment (fivetran-joemarkiewicz) [inline macros/timestamp_diff.sql:8]:
```suggestion
    datediff(
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
lgtm - one small change

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM!

**Pull request merged**